core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/aggregate/AggregateOpDesc.scala

```diff
@@ -16,9 +16,13 @@ import edu.uci.ics.texera.workflow.common.metadata.{OperatorGroupConstants, Oper
 import edu.uci.ics.texera.workflow.common.operators.LogicalOp
 import edu.uci.ics.texera.workflow.common.workflow.HashPartition
 
+import javax.validation.constraints.{NotNull, Size}
+
 class AggregateOpDesc extends LogicalOp {
   @JsonProperty(value = "aggregations", required = true)
   @JsonPropertyDescription("multiple aggregation functions")
+  @NotNull(message = "aggregation cannot be null")
+  @Size(min = 1, message = "aggregations cannot be empty")
```

Comment:
Why do we need size here?

Reply:
aggregations requires @Size(min = 1) because it’s a list. @NotNull ensures it’s not null, @Size(min = 1) ensures the list has elements.